File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) <= 1:
            return len(s)

        chars = set()
        l = r = 0
        res = 0
        while l <= r and r < len(s):
            while s[r] in chars:
                chars.remove(s[l])
                l += 1
            else:
                res = max(res, r - l + 1)
                chars.add(s[r])
                r += 1
        return res

File: Data Structures & Algorithms/longest-strictly-increasing-or-strictly-decreasing-subarray/submission-2.rs
impl Solution {
    pub fn longest_monotonic_subarray(nums: Vec<i32>) -> i32 {
        let mut last: Option<i32> = None;
        let mut res = 0;
        let mut count = 0;

        for i in 0..nums.len() {
            if last.is_some() && last.unwrap() >= nums[i] {
                count = 0;
            }
            count += 1;
            last = Some(nums[i]);
            res = res.max(count);
        }

        count = 0;

        for i in 0..nums.len() {
            if  last.is_some() && last.unwrap() <= nums[i] {
                count = 0;
            }
            count += 1;
            last = Some(nums[i]);
            res = res.max(count);
        }

        res
    }
}
